packages/server/src/tests/header/headerSelection.ts

```diff
@@ -107,91 +91,65 @@ const supportersTest: HeaderTest = {
 
-const isInUsEoyPeriod = (date: Date): boolean => {
-    return isAfter(date, usEoyPeriodStart) && isBefore(date, usEoyPeriodEnd);
-};
+    const selectedTest: HeaderTest | undefined = allTests.find(test => {
+        const { isOn, userCohort, locations } = test;
+
+        if (!isOn) {
+            return false;
+        }
+
+        if (showSupportMessaging && !isNotReportedAsSupporter.includes(userCohort)) {
+            return false;
+        }
 
-const hasNotContributedDuringUsEoyPeriod = (lastOneOffContributionDate?: string): boolean => {
-    if (!lastOneOffContributionDate) {
+        if (!showSupportMessaging && isNotReportedAsSupporter.includes(userCohort)) {
+            return false;
+        }
+
+        if (!locations.includes(countryGroupId)) {
+            return false;
+        }
         return true;
+    });
+
+    // Failed to find a matching test, or the matching test has an empty variants Array
+    if (!selectedTest || !selectedTest.variants.length) {
+        return null;
     }
 
-    return isBefore(new Date(Date.parse(lastOneOffContributionDate)), usEoyPeriodStart);
-};
+    const selectedVariant: HeaderVariant = selectVariant(selectedTest, targeting.mvtId);
 
-const getNonSupportersTest = (edition: Edition): HeaderTest => {
-    if (edition === 'UK') {
-        return nonSupportersTestUK;
-    }
-    if (edition === 'US' && isInUsEoyPeriod(new Date())) {
-        return nonSupportersTestUS();
-    }
-    return nonSupportersTestNonUK;
-};
+    selectedVariant.modulePathBuilder = modulePathBuilder;
 
-const getSupportersTest = (edition: Edition, lastOneOffContributionDate?: string): HeaderTest => {
-    if (
-        edition === 'US' &&
-        isInUsEoyPeriod(new Date()) &&
-        hasNotContributedDuringUsEoyPeriod(lastOneOffContributionDate)
-    ) {
-        return supportersTestUS;
-    }
-    return supportersTest;
+    return {
+        test: selectedTest,
+        variant: selectedVariant,
+        moduleName: header.name,
+        modulePathBuilder,
+    };
 };
 
 export const selectHeaderTest = (
     targeting: HeaderTargeting,
 ): Promise<HeaderTestSelection | null> => {
-    const select = (): HeaderTest => {
-        if (targeting.showSupportMessaging) {
-            return getNonSupportersTest(targeting.edition);
-        } else {
-            return getSupportersTest(targeting.edition, targeting.lastOneOffContributionDate);
-        }
-    };
-
-    const test = select();
-    const variant = test.variants[targeting.mvtId % test.variants.length];
-    if (test && variant) {
-        return Promise.resolve({
-            test,
-            variant,
-            moduleName: header.name,
-            modulePathBuilder: variant.modulePathBuilder,
-        });
-    }
-    return Promise.resolve(null);
+    return fetchConfiguredHeaderTestsCached()
+        .then((allTests: HeaderTest[]) => {
+            if (allTests == null) {
```

Comment:
`allTests` would never be null here, as `fetchConfiguredHeaderTestsCached` returns an array.
You can just do e.g. 
`const allTests = [...configuredTests, supportersTest, nonSupportersTestUK, nonSupportersTestNonUK]`

Reply:
Also the catch can return `[]` instead of `null` if we decouple the fetch (which can go wrong) from the construction of the array with the fallback tests:
```
const configuredTests = await fetchConfiguredHeaderTestsCached().catch(() => []);
return selectBestTest(targeting, [...configuredTests, ...hardcodedTests]);
```

Reply:
I prefer the catch to be specific to the fetch, which can fail.
The construction of the tests array, and the calling of `selectBestTest`, are not side-effecting.

Reply:
So the case is that if the fetch fails, then our preference is to show no content in the header space, rather than to show default hardcoded header data? Or does the system add something to the header space from elsewhere if the fetch leads to a null (or []) result?

Reply:
If you use my above suggestion then it will always use hardcoded tests, regardless of whether the fetch fails